### ISSUE TITLE
Feature request: Add a "zoom" function

### ISSUE THREAD

**Issue by daattali** [almende/vis#1955]
I discovered this library today and it is absolutely amazing. Really great job.  I do think that a zoom function is missing, it'd be great to programmatically be able to tell the timeline to zoom in or out. 

I did find an old issue https://github.com/almende/vis/issues/143 where there is some code showing how to implement that, but it would be great to have that built-in as a simple API we could call.

**Comment by daattali** [comment]
The code to do the zooming is actually not available anymore in the current git HEAD and I had to dig around in the git history to find the right file and function, so it might benefit others if I post it:

```
    function zoom (percentage) {
        var range = timeline.getWindow();
        var interval = range.end - range.start;

        timeline.setWindow({
            start: range.start.valueOf() - interval * percentage,
            end:   range.end.valueOf()   + interval * percentage
        });
    }
```

**Comment by mojoaxel** [comment]
There is an example that shows manual zooming: [navigationMenu.html](https://github.com/almende/vis/blob/master/examples/timeline/interaction/navigationMenu.html) ([live](http://visjs.org/examples/timeline/interaction/navigationMenu.html))

**Comment by daattali** [comment]
Thanks @mojoaxel , I didn't notice that.

The zoom implemented there is kind of dumb (for example, zooming in and then out does not end up in the same place as where you started).  For anyone who wants a smarter zoom, I worked out the math to do it on paper, and here's a zoomIn and zoomOut functions that work nicely.  For example, starting with a window of [2010-01-01, 2010-05-05] and then zooming out 5 times and then zooming in 5 times will end up exactly with the original window.

```
     zoomIn : function(percentage, animation) {
        if (typeof animation === "undefined") {
          animation = true;
        }
        var range = timeline.getWindow();
        var start = range.start.valueOf();
        var end = range.end.valueOf();
        var interval = end - start;
        var newInterval = interval / (1 + percentage);
        var distance = (interval - newInterval) / 2;
        var newStart = start + distance;
        var newEnd = end - distance;

        timeline.setWindow({
          start   : newStart,
          end     : newEnd,
          animation : animation
        });
      },
      zoomOut : function(percentage, animation) {
        if (typeof animation === "undefined") {
          animation = true;
        }
        var range = timeline.getWindow();
        var start = range.start.valueOf();
        var end = range.end.valueOf();
        var interval = end - start;
        var newStart = start - interval * percentage / 2;
        var newEnd = end + interval * percentage / 2;

        timeline.setWindow({
          start   : newStart,
          end     : newEnd,
          animation : animation
        });
      },
```

**Comment by daattali** [comment]
I opened PR https://github.com/almende/vis/pull/1986 if you want to incorporate that zoom into the existing example

**Comment by daattali** [comment]
So has this been implemented, or is it not going to happen? I see the issue was closed but no comment was made

**Comment by yotamberk** [comment]
I thought it was implemented with your PR. My mistake.

**Comment by mojoaxel** [comment]
This was fixed with adedd6a853c33184ab7c819eea1815a791267e8c

**Comment by daattali** [comment]
thank you @mojoaxel

**Comment by daattali** [comment]
Can you consider adding "zoomin/zoomout" as API methods of the timeline, rather than just having this as an example?

**Comment by mojoaxel** [comment]
> Can you consider adding "zoomin/zoomout" as API methods of the timeline, rather than just having this as an example?

Of course! Can you please create a new issue for that. Also feel free to provide a pull-request (this time to the develop branch, please).

**Comment by daattali** [comment]
Are you sure I should create a new issue vs reopening this one? If you look at this issue'so title and description, it was describing exactly this. That's why I was confused when the issue got closed. This issue specifically asks for adding zoom as a feature of the library.

**Comment by mojoaxel** [comment]
@daattali Of course you are right.
We should remove the zoom functions from the example and move the functions to the lib.

**Comment by mojoaxel** [comment]
fixed with #2239

**Comment by daattali** [comment]
Awesome, thank you!
